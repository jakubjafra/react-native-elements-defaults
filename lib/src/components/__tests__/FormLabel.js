const testHelperUtils = require('../../testHelperUtils');

const utils = require('../../utils');
const Overwrite = utils.Overwrite;

const name = 'FormLabel',
    __oldElement__ = require('react-native-elements/src/form/FormLabel'),
    __element__ = require('../FormLabel'),
    noTheme = new Overwrite(
        name,
        Overwrite.TYPE.CLASS_COMPONENT_ATTRIBUTES,
        Overwrite.SCOPE.DEFAULTS,
        {}
    ),
    basicTheme = new Overwrite(
        name,
        Overwrite.TYPE.CLASS_COMPONENT_ATTRIBUTES,
        Overwrite.SCOPE.DEFAULTS,
        {
            containerStyle: {
                backgroundColor: 'red',
            }
        }
    );

describe(name, () => {
    testHelperUtils.defaultTests(name, __oldElement__, __element__, noTheme, basicTheme);
});
