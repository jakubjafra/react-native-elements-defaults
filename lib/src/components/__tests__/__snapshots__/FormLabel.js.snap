// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel with basic theme applied FormLabel renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 15,
          },
          undefined,
          undefined,
        ],
      ]
    }
  />
</View>
`;

exports[`FormLabel with basic theme applied OldFormLabel renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 15,
          },
          undefined,
          undefined,
        ],
      ]
    }
  />
</View>
`;

exports[`FormLabel with no theme applied FormLabel renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 15,
          },
          undefined,
          undefined,
        ],
      ]
    }
  />
</View>
`;

exports[`FormLabel with no theme applied OldFormLabel and FormLabel returns the same thing if no theme is applied 1`] = `"[2mCompared values have no visual difference.[22m"`;

exports[`FormLabel with no theme applied OldFormLabel renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 15,
          },
          undefined,
          undefined,
        ],
      ]
    }
  />
</View>
`;
